modules/standard/Set.chpl

```diff
@@ -329,9 +341,16 @@ module Set {
     */
     proc ref clear() {
       on this {
-        _enter();
-        _dom.clear();
-        _leave();
+        _enter(); defer _leave();
+        for idx in 0..#_htb.tableSize {
+          pragma "no init"
```

Comment:
You can remove this pragma like this:

``` chapel
if _htb.isSlotFull(idx) {
  var key: eltType;
  var val: nothing;
  _htb.clearSlot(idx, key, val);
}
```

split init with out intent should remove the need for the pragma.

Reply:
Ok, thanks!